Issue #16332: Updated SuppressionSingleFilterExamplesTest methods to use verifyFilterWithInlineConfigParser

Prarthana-Ramesh opened the pull request:
Issue: #16332 

**Example2.java ->**
`    // filtered violation below 'String literal expressions should be on the left side of an equals comparison.'
`
Contains a filtered violation that exceeds the line length limit of 85. To address this, it has been added to checkstyle-examples-suppressions.xml.

**Example3.java ->**
`      "This is an example string." // filtered violation 'Line matches the illegal pattern 'example''`
Requires the filtered violation comment to be on the same line as the violation itself to prevent the word "example" in the comment from being incorrectly flagged. This results in a line length exceeding 85, so it has been included in checkstyle-examples-suppressions.xml.

**Example9.java ->**
```
Violation lines for C:\Users\rprar\OneDrive\Desktop\checkstyle\src\xdocs-examples\resources\com\puppycrawl\tools\checkstyle\filters\suppressionsinglefilter\Example9.java differ.
value of: iterable.onlyElement()
expected: 17
but was : 1
Expected :17
Actual   :1
```
Triggers a file length violation, which always starts at line 1. However, since line 1 must contain XML configuration, adding a filtered violation comment isn't possible. To resolve this, the maximum file length has been set higher than the actual file length, effectively preventing a violation in the file.

Please let me know if any corrections or refinements are to be made.

romani reviewed (changes requested):
Items

romani commented on config/checkstyle-examples-suppressions.xml:
```diff
@@ -143,6 +143,11 @@
   <suppress checks="LineLength"
     files="linelength[\\/]Example\d+.*"/>
+  <suppress checks="LineLength"
+    files="suppressionsinglefilter[\\/]Example3.java"/>
```
Not ok, please wrap lines, or use violation above|below.

romani reviewed (changes requested):
Items

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/suppressionsinglefilter/Example9.java:
```diff
@@ -14,6 +14,6 @@
 // xdoc section -- start
 public class Example9 {
-  //filtered violation 'File length is 19 lines (max allowed is 1)'
+
```
Revert your changes on this file.
Put `filtered violation 14 lines above  'File length is 19 lines (max allowed is 1)'
Before `// xdoc section -- start`

Change line 17 to `/* filtered violation on 1st line  'File length is 19 lines (max allowed is 1)' */`
To not be considered as marker for violation, but be visible in examples html page

romani replied:
not done.

romani replied:
please make it `/* filtered violation on 1st line  'File length is 19 lines (max allowed is 1)' */`
in  this case it will not be considered as violation marker and will be just comment.

and violation marker comment you put above xdoc start comment

like:
```
// filtered violation 16 lines above 'File length is 20 lines (max allowed is 1)'
// xdoc section -- start
```

Prarthana-Ramesh replied:
Changes updated accordingly. Please review.

romani reviewed (changes requested):
Items

Prarthana-Ramesh commented:
I've updated _Example3.java_ and removed the suppression from `checkstyle-examples-suppressions.xml`.

However, regarding _Example9.java_, I found `// violation x lines above` in
[InlineConfigParser.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java), but there's no mention of `// filtered violation x lines above`. As a result, the violation isn't being filtered.
Could you please guide me on how to resolve this?

romani commented:
> Contains a filtered violation that exceeds the line length limit of 85. To address this, it has been added to checkstyle-examples-suppressions.xml.

`// filtered violation below 'String literal expressions should be on the left side'`

You can cut message to avoid suppression. Please avoid suppression

Prarthana-Ramesh commented:
All suppressions removed, please check and let me know if any further modifications are required.

romani commented:
GitHub, generate website

romani commented on config/checkstyle-examples-suppressions.xml:
```diff
@@ -143,6 +143,7 @@
     files="suppresswithplaintextcommentfilter[\\/]Example9.java"/>
   <suppress checks="LineLength"
     files="linelength[\\/]Example\d+.*"/>
+
```
please revert this change

Prarthana-Ramesh replied:
change reverted.

romani commented on src/site/xdoc/filters/suppressionsinglefilter.xml:
```diff
@@ -129,8 +129,7 @@
         <div class="wrapper"><pre class="prettyprint"><code class="language-java">
 public class Example1 {
   public void exampleMethod() {
```
please help to fix description above example

from `reject <code>JavadocStyleCheck</code> violations for lines 82 and 108 to 122 of`
to actual . 
Whole description should be updated to reflect what is in config.

Prarthana-Ramesh replied:
Descriptions updated according to config. Please review and let me know if changes are required.

romani replied:
please update https://github.com/checkstyle/checkstyle/blob/master/src/site/xdoc/filters/suppressionsinglefilter.xml.template to have proper description

Prarthana-Ramesh replied:
Sorry for the inconvenience. It is updated now.

romani commented on src/site/xdoc/filters/suppressionsinglefilter.xml:
```diff
@@ -140,12 +139,17 @@ public class Example1 {
+  &lt;/module&gt;
+  &lt;module name="SuppressionSingleFilter"&gt;
+    &lt;property name="files" value="Example2.java"/&gt;
+    &lt;property name="checks" value="JavadocMethod"/&gt;
+  &lt;/module&gt;
   &lt;module name="SuppressionSingleFilter"&gt;
-    &lt;property name="id" value="stringEqual"/&gt;
     &lt;property name="files" value="Example2.java"/&gt;
-    &lt;property name="checks" value="EqualsAvoidNull, JavadocMethod"/&gt;
+    &lt;property name="checks" value="EqualsAvoidNull"/&gt;
```
can we reuse this form `property name="checks" value="EqualsAvoidNull, JavadocMethod"` ?

romani replied:
https://checkstyle.sourceforge.io/filters/suppressionsinglefilter.html#Properties

>checks 	**Define the RegExp** for matching against the name of the check associated with an audit event.

so please do:
`property name="checks" value="EqualsAvoidNull|JavadocMethod"`

romani replied:
please do this in all example where you have two/few Checks.

Prarthana-Ramesh replied:
Done. Changes made to all example files with multiple checks.

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/suppressionsinglefilter/Example2.java:
```diff
@@ -1,21 +1,26 @@
-  public void checkStringEquality(String str1, String str2) {
-    // filtered violation ''.equals()' should be used for string comparison'
-    assert str1 == str2 ;
+  public void checkStringEquality(String s) {
```
why there is no comment on filtered violation for JavadocMethod ?

Prarthana-Ramesh replied:
Comment on filtered violation for JavadocMethod added.

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/suppressionsinglefilter/Example3.java:
```diff
@@ -1,7 +1,7 @@
 <module name="Checker">
   <module name="RegexpSingleline">
-    <property name="format" value=".*example.*"/>
+    <property name="format" value="&#101;&#120;&#97;&#109;&#112;&#108;&#101;"/>
```
please revert and add violation comment line above `// xdoc section -- start`
so it will not be visible to users, and yes it is expected that it mathing this line.

Prarthana-Ramesh replied:
Reverting to the encoded format `(&#101;&#120;&#97;&#109;&#112;&#108;&#101;)` will cause the XML configuration line itself to match the RegexpSingleline rule of `'example'`, leading to a detected violation. However, this violation cannot be filtered since the` // filtered violation` comment cannot be added directly within the XML configuration. Additionally, attempting to suppress it using` // filtered violation x lines above` would introduce a new violation at the comment line, making suppression ineffective. Please guide me through this.

romani replied:
you need to put above `// xdoc section -- start`

line like `// filtered violation X lines above 'Line matches the illegal pattern'` 
as it above `// xdoc section -- start` , it will not be in documentation so user will not see it, 
attention message is cut in violation comment to not match one more time

Prarthana-Ramesh replied:
Understood. Changes updated accordingly. Please review.

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/suppressionsinglefilter/Example7.java:
```diff
@@ -1,20 +1,23 @@
 /*xml
 <module name="Checker">
   <module name="MemberName"/>
+  <module name="MethodName"/>
```
please add violation for it, to show when it is filtered.

Prarthana-Ramesh replied:
Violation added for MethodName.

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/suppressionsinglefilter/Example9.java:
```diff
@@ -2,7 +2,7 @@
   <module name="TreeWalker">
     <module name="FileLength">
-      <property name="max" value="1"/>
+      <property name="max" value="19"/>
```
revert this line, see somment above

Prarthana-Ramesh replied:
The max value was reverted to 1 after modifying InlineConfigParser.java to support `// filtered violation x lines above` comments. Previously, it was set to 19 to prevent violations due to the lack of this suppression feature.

romani reviewed (changes requested):
items:

romani commented on src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilterExamplesTest.java:
```diff
@@ -31,91 +31,114 @@ protected String getPackageLocation() {
-        final String[] expected = {
-
+        final String[] expectedWithoutFilter = {
+            "23:13: String literal expressions should be on the left side of an equals comparison.",
```
can we user message variables here to avoid copy paste of messages.
See example at https://github.com/checkstyle/checkstyle/blob/82c21d5175fd1c5d39edaff6a79c726d7d2a7861/src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilterTest.java#L86

Prarthana-Ramesh replied:
All message variables updated to the above format. Please review.

romani reviewed (changes requested):
items:

romani commented:
GitHub, generate website

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/suppressionsinglefilter/Example3.java:
```diff
@@ -10,19 +10,19 @@
   public void noViolation() {
     System.out.println(
-      "This string does not contain 'example'."
+      "This string does not contain 'Example'."
```
``"RegexpSingleline is case sensitive by default. 'Example' in not matching."``

it is hard to catch that https://checkstyle.org/checks/regexp/regexpsingleline.html#RegexpSingleline is case sensitive by default

Prarthana-Ramesh replied:
do you suggest that I change it back to ->

`"This string does not contain 'example'."`

OR

retain the same and add comment
`// ignoreCase (defaultValue) - false`

OR

change ignoreCase to true and add a `new violation comment`

Prarthana-Ramesh replied:
done.

romani reviewed (changes requested):
thanks a lot !!!

last:

Prarthana-Ramesh commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/suppressionsinglefilter/Example9.java:
```diff
@@ -12,8 +12,10 @@
 package com.puppycrawl.tools.checkstyle.filters.suppressionsinglefilter;
+// filtered violation 15 lines above 'File length is 21 lines (max allowed is 1)'
 // xdoc section -- start
+/* filtered violation on 1st line  'File length is 21 lines (max allowed is 1)' */
```
done.

romani commented:
GitHub, generate website

romani commented:
Please put in commit all changes after test execution on local.
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=27923&view=logs&j=d72e04f6-d294-5bfe-b3cd-ffcb523f40f9&t=a021e897-39dd-596d-375e-f941387b47d5&l=1964

romani commented:
GitHub, generate website

romani reviewed (approved):
Ok to merge.

Thanks a lot , it was huge work , but you completed it 
Now all examples are actual and correct, so users will appreciate it for sure

romani commented:
https://github.com/checkstyle/checkstyle/issues/16729
Will be follow up, this PR is good step forward, let's merge it.

Prarthana-Ramesh commented:
Thank you, learnt a lot in the process. Looking forward to contribute more. Can I work on the follow up issue as well?

romani commented:
@Prarthana-Ramesh , sure !!! please be welcome.
